src/plugins/dashboard/public/application/lib/build_dashboard_container.ts

```diff
@@ -109,9 +109,11 @@ export const buildDashboardContainer = async ({
         gridData: originalPanelState.gridData,
         type: incomingEmbeddable.type,
         explicitInput: {
-          ...(incomingEmbeddable.type === originalPanelState.type && {
-            ...originalPanelState.explicitInput,
-          }),
+          ...(incomingEmbeddable.type === originalPanelState.type
+            ? {
+                ...originalPanelState.explicitInput,
+              }
+            : { hidePanelTitles: originalPanelState.explicitInput.hidePanelTitles }),
```

Comment:
I'm curious about this change, as the editor is in charge of returning the full input, and `hidePanelTitles` is part of the input, you could restore it by ensuring that the value isn't deleted between the Dashboard -> TSVB -> Lens navigation.

This change likely works okay, but it seems a little poorly defined from the code side to say that the `hidePanelTitles` should never update on `save and return` if the new embeddable's type is different.

If we eventually allowed editing of the title and the `hidePanelTitles` key from inside the editor, we would likely expect changes there to be applied, not overwritten.

If at all possible, I would encourage you to try to maintain the `hidePanelTitles` value instead.

Reply:
In that case I will need to put this setting in navigate to lens config, then get this when user click on "Replace in dashboard" and define it in new embeddable input on Lens side. For me it looks strange that Lens will know about specific dashboard panel setting  - `hidePanelTitles` and only when we use conversion logic

Reply:
These are weird settings to be honest, `hidePanelTitles`, and `title` are dashboard implementation details, but they are stored in `explicitInput` which means they should be managed on the individual embeddable level. It isn't ideal, but that is how the architecture is at the moment.

That said, I'm okay if you want to leave this here in order to make the implementation on your side simpler, but it would be nice to have a comment at least explaining that this is temporary, and only required because the key is stored in `explicitInput` when it should be stored outside of it instead.

On our side, we could look into moving this to a better place.

Reply:
Added comment for that